components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -112,7 +112,7 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
 
         if (StringUtils.isEmpty(userStoreDomain)) {
             context.setSubject(
-                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserName));
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(fullyQualifiedName));
```

Comment:
Hmm... sounds a fair requirement. There is a small catch. IWA authenticator can work in two modes. 

1. Local Authenticator
2. Federated Authenticator

In case 1 we check for the existence of the user in user stores of Identity Server. '@' is usually not allowed in a username unless we configure Identity Server for email usernames. 
So if we want to pass username with the Kerberos domain we could do so by enabling email usernames in this mode. Plus we also need the fully qualified username to be in the userstores of the identity server.

In the usual case what happens in let's say user1@domain1.com is authenticated, we check for the existence of user1 in user stores configured for domain1 when defining the IDP.



In case2 there won't be a problem.
If you could let's start a discussion on dev@wso2.org. Otherwise